_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,519 @@
+---
+layout: default
+title: Search with k-NN filters
+nav_order: 10
+parent: k-NN
+has_children: false
+has_math: true
+---
+
+# Search for k-NN with filters
+Introduced 2.4
+{: .label .label-purple }
+
+You can create custom filters using Query DSL search options to refine your k-NN searches. You define the filter criteria within the `knn_vector` field's `filter` subsection in your query. You can use all available Query DSL query types such as: term, range, regex and wildcard. To include or exclude results, you specify Boolean query clauses. You also specify a query point with the `knn_vector` type and search for nearest neighbors that match your filter criteria.
+To run k-NN queries with a filter, the Lucene search engine and HSNW method are required.
+
+To learn more about how to use Query DSL Boolean query clauses, see [Boolean queries]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/bool).
+{: .note }
+
+## How does a k-NN filter work?
+
+Starting from OpenSearch version 2.2 the k-NN plugin uses the Lucene engine that provides a search process for filtered documents that uses a Hierarchical Navigable Small World (HSNW) algorithm to represent a multi-layered graph to filter searches. After a filter is applied to a set of documents to be searched, the algorithm decides whether to perform pre-filtering for an exact kNN search or modified post-filtering for approximate search. The approximate search with filtering guarantees the top number of closest vectors in result.
```

Comment:
I think first sentence is a bit misleading in terms of version and products. Lucene introduced knn search in 9.0 and filtering for this search in 9.1. On our end we released lucene based knn in 2.2 (at this point core OpenSearch and knn plugin were using Lucene 9.2). Filtering feature in knn plugin will be released in version 2.4 and it uses Lucene 9.4. 
I think we need to change sentence to something like, please feel free to change exact wording as needed:
```
OpenSearch k-NN plugin uses Lucene engine for search process starting from version 2.2. Lucene engine provides a search that is based on Hierarchical Navigable Small World (HSNW) algorithm to represent a multi-layered graph. From OpenSearch version 2.4 k-NN plugin is able to incorporate filters in search, that is based on Lucene 9.4.
```
and then continue about the filtering algorithm.